SAMException when TRUNCATE_NAMES_AT_WHITESPACE=False in CreateSequenceDictionary

The following example fasta file causes an exception if TRUNCATE_NAMES_AT_WHITESPACE is set to false.  This option should possibly be removed.

```
> chr1 12lk
ACGTACGT
> chr2 567
TGCATCGA
```

```
Caused by: htsjdk.samtools.SAMException: Sequence name contains invalid character: chr1 12lk
    at htsjdk.samtools.SAMSequenceRecord.<init>(SAMSequenceRecord.java:85)
    at org.broadinstitute.hellbender.tools.picard.CreateSequenceDictionary.makeSequenceRecord(CreateSequenceDictionary.java:104)
    at org.broadinstitute.hellbender.tools.picard.CreateSequenceDictionary.makeSequenceDictionary(CreateSequenceDictionary.java:95)
    at org.broadinstitute.hellbender.tools.picard.CreateSequenceDictionary.doWork(CreateSequenceDictionary.java:71)
    at org.broadinstitute.hellbender.cmdline.CommandLineProgram.instanceMain(CommandLineProgram.java:148)
    at org.broadinstitute.hellbender.cmdline.PicardCommandLineProgram.instanceMain(PicardCommandLineProgram.java:51)
    at org.broadinstitute.hellbender.Main.instanceMain(Main.java:101)
    at org.broadinstitute.hellbender.Main.instanceMain(Main.java:108)
    at org.broadinstitute.hellbender.CommandLineProgramTest.runCommandLine(CommandLineProgramTest.java:55)
    at org.broadinstitute.hellbender.tools.picard.CreateSequenceDictionaryTest.testNonUniqueSequenceName(CreateSequenceDictionaryTest.java:50)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:483)
    at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:84)
    at org.testng.internal.Invoker.invokeMethod(Invoker.java:714)
    ... 24 more
```

--- a/src/main/java/org/broadinstitute/hellbender/tools/picard/CreateSequenceDictionary.java
+++ b/src/main/java/org/broadinstitute/hellbender/tools/picard/CreateSequenceDictionary.java
@@ -43,10 +43,6 @@ public class CreateSequenceDictionary extends PicardCommandLineProgram {
     @Option(doc = "Put into SP field of sequence dictionary entry", optional = true)
     public String SPECIES;
 
-    @Option(doc = "Make sequence name the first word from the > line in the fasta file.  " +
-            "By default the entire contents of the > line is used, excluding leading and trailing whitespace.")
-    public boolean TRUNCATE_NAMES_AT_WHITESPACE = true;
-
     @Option(doc = "Stop after writing this many sequences.  For testing.")
     public int NUM_SEQUENCES = Integer.MAX_VALUE;
 
@@ -83,7 +79,7 @@ public class CreateSequenceDictionary extends PicardCommandLineProgram {
      */
     SAMSequenceDictionary makeSequenceDictionary(final File referenceFile) {
         final ReferenceSequenceFile refSeqFile =
-                ReferenceSequenceFileFactory.getReferenceSequenceFile(referenceFile, TRUNCATE_NAMES_AT_WHITESPACE);
+                ReferenceSequenceFileFactory.getReferenceSequenceFile(referenceFile, true);
         ReferenceSequence refSeq;
         final List<SAMSequenceRecord> ret = new ArrayList<SAMSequenceRecord>();
         final Set<String> sequenceNames = new HashSet<String>();

--- a/src/test/java/org/broadinstitute/hellbender/tools/picard/CreateSequenceDictionaryTest.java
+++ b/src/test/java/org/broadinstitute/hellbender/tools/picard/CreateSequenceDictionaryTest.java
@@ -29,7 +29,6 @@ public class CreateSequenceDictionaryTest extends CommandLineProgramTest {
         final String[] argv = {
                 "REFERENCE=" + BASIC_FASTA,
                 "OUTPUT=" + outputDict,
-                "TRUNCATE_NAMES_AT_WHITESPACE=false"
         };
         Assert.assertEquals(runCommandLine(new ArgumentsBuilder(argv).getArgsArray()), null);
     }
@@ -45,7 +44,6 @@ public class CreateSequenceDictionaryTest extends CommandLineProgramTest {
         final String[] argv = {
                 "REFERENCE=" + DUPLICATE_FASTA,
                 "OUTPUT=" + outputDict,
-                "TRUNCATE_NAMES_AT_WHITESPACE=false"
         };
         Assert.assertEquals(runCommandLine(new ArgumentsBuilder(argv).getArgsArray()), null);
         Assert.fail("Exception should have been thrown.");

